plugins/chat/spec/system/list_channels/drawer_spec.rb

```diff
@@ -108,19 +119,78 @@
 
     context "when multiple channels are present" do
       fab!(:user_1) { Fabricate(:user) }
+      fab!(:user_2) { Fabricate(:user) }
+      fab!(:user_3) { Fabricate(:user) }
       fab!(:dm_channel_1) { Fabricate(:direct_message_channel, users: [current_user]) }
       fab!(:dm_channel_2) { Fabricate(:direct_message_channel, users: [current_user, user_1]) }
+      fab!(:dm_channel_3) { Fabricate(:direct_message_channel, users: [current_user, user_2]) }
+      fab!(:dm_channel_4) { Fabricate(:direct_message_channel, users: [current_user, user_3]) }
+
+      it "sorts them by latest activity" do
+        drawer_page.visit_index
+        drawer_page.click_direct_messages
 
-      before do
         Fabricate(:chat_message, chat_channel: dm_channel_2, user: user_1, use_service: true)
+
+        # last message was read but the created at date is used for sorting
+        message =
+          Fabricate(:chat_message, chat_channel: dm_channel_4, user: user_3, use_service: true)
+        dm_channel_4.membership_for(current_user).mark_read!(message.id)
+
+        expect(drawer_page).to have_channel_at_position(dm_channel_2, 1)
+        expect(drawer_page).to have_urgent_channel(dm_channel_2)
+
+        expect(drawer_page).to have_channel_at_position(dm_channel_4, 2)
+        expect(drawer_page).to have_channel_at_position(dm_channel_1, 3)
+        expect(drawer_page).to have_channel_at_position(dm_channel_3, 4)
       end
 
-      it "sorts them by latest activity" do
+      it "sorts channels with threads by urgency" do
         drawer_page.visit_index
         drawer_page.click_direct_messages
 
-        expect(drawer_page).to have_channel_at_position(dm_channel_2, 1)
-        expect(drawer_page).to have_channel_at_position(dm_channel_1, 2)
+        # watched thread
```

Comment:
I would prefer you use variables names instead of comments

Reply:
I've added a new transient to the fabricator to set the OP membership level, which cleans this up a bit